RFC-0039-cuda-support.md

```diff
@@ -0,0 +1,197 @@
+
+4. Update CUDA in CI
+  When: Update CUDA in CD is complete
+  Goal: Make sure CI jobs tests on new CUDA versions, all failing tests should be identified and reported
+
+5. Run benchmarks to compare new experimental CUDA version to current stable CUDA version
+  When: Update CUDA in CD  is complete
+  Goal: Make sure we identify all performance regressions
+
+
+### Detailed Process of Deprecating CUDA version
+
+1. Evaluate deprecation of legacy CUDA version from CI/CD
+When: We completed CUDA update and previous experimental CUDA version is qualified to be stable and we have 3 supported versions (legacy, stable and experimental). Start by creating RFC [issue](https://github.com/pytorch/pytorch/issues/147383) to deprecate legacy CUDA version from CI/CD.
+Goal: Make sure we support 2 versions of CUDA, supporting 3 versions can be an exception for certain release where we need to keep legacy version
+
+2. Deprecate legacy CUDA version from CI/CD
+When: Evaluate deprecation of legacy CUDA version from CI/CD is complete
+Goal: Support for legacy CUDA versions is dropped, starting from PyTorch Domain Libraries and then in PyTorch core. First we drop CD support and then CI support.
+
+
+# CUDA/CUDNN Upgrade Runbook
+
+So you wanna upgrade PyTorch to support a new CUDA? Follow these steps in order! They are adapted from previous CUDA upgrade processes.
+
+## 0. Before starting upgrade process
+
+### A. Currently Supported CUDA and CUDNN Libraries
+Here is the supported matrix for CUDA and CUDNN (versions can be looked up in [here](https://github.com/pytorch/pytorch/blob/main/.ci/docker/common/install_cudnn.sh) . Architecture support can be found [here](https://github.com/pytorch/pytorch/blob/main/.ci/manywheel/build_cuda.sh))
+
+| CUDA | CUDNN | architectures supported | additional details |
+| --- | --- | --- | --- |
+| 11.8.0 | 9.1.0.70 | Kepler(3.7), Maxwell(5.0), Pascal(6.0), Volta(7.0), Turing(7.5), Ampere(8.0, 8.6), Hopper(9.0) | Legacy CUDA Release |
+| 12.6.3 | 9.5.1.17 | Maxwell(5.0), Pascal(6.0), Volta(7.0), Turing(7.5), Ampere(8.0, 8.6), Hopper(9.0)  | Stable CUDA Release |
+| 12.8.0 | 9.7.1.26 | Turing(7.5), Ampere(8.0, 8.6), Hopper(9.0), Blackwell(10.0, 12.0+PTX)  | Latest CUDA Release |
+|        | 9.8.0.87 | Turing(7.5), Ampere(8.0, 8.6), Hopper(9.0), Blackwell(10.0, 12.0+PTX)  | Latest CUDA Nightly |
+
+### B. Check the package availability
+
+Package availability to validate before starting upgrade process :
+
+1) CUDA and CUDNN is available for Linux and Windows:
+https://developer.download.nvidia.com/compute/cuda/12.8.0/local_installers/cuda_12.8.0_570.86.10_linux.run (x86)
+https://developer.download.nvidia.com/compute/cuda/12.8.0/local_installers/cuda_12.8.0_570.86.10_linux_sbsa.run (aarch64)
+https://developer.download.nvidia.com/compute/cudnn/redist/cudnn/
+
+2) CUDA is available on Docker hub images : https://hub.docker.com/r/nvidia/cuda
```

Comment:
Do we really need this? I thought we moved to plain Linux containers and are using our [`install_cuda.sh`](https://github.com/pytorch/pytorch/blob/0959869683349f83a829741f010de775d2072e18/.ci/docker/common/install_cuda.sh#L13) script?

Reply:
I believe this is still used, however should be deprecated soon